### PR TITLE
gguf-py : export chat templates

### PR DESCRIPTION
**Pull request by slaren**
Exports the chat templates from HF `tokenizer_config.json` as `tokenizer.chat_template`. 

Chat templates are not supported by llama.cpp, but 3rd party applications can use `llama_model_meta_val_str` or any of the GGUF libraries to access the template metadata.

### CODE REVIEW

**Comment by slaren** [comment]
I am not very familiar with the gguf-py code, so I would appreciate any comments. Do we need to bump the gguf-py version in `pyproject.toml`?

File: gguf-py/pyproject.toml
Comment (KerfuffleV2) [inline gguf-py/pyproject.toml:3]: We do want to bump it (so people who update the package get the new version). If it was me, I would just bump it to `0.5.4` though.

I am not 100% sure that's correct so it would absolutely be reasonable to get a second opinion from someone like cebtenzzre who could give you a definitive answer.

Comment (slaren) [reply gguf-py/pyproject.toml:3]: I was going by the definition of [semantic versioning](https://semver.org/), where the version numbers are given by `MAJOR.MINOR.PATCH`. Since this is not a bug fix, bumping the minor version seemed the right thing to do here.

File: gguf-py/gguf/vocab.py
Comment (KerfuffleV2) [inline gguf-py/gguf/vocab.py:164]: You probably also need to add some logic to check that it's the correct type before you put it in `self.chat_template`. Otherwise you could put an object or a number or something random in there. I didn't check yet, but I'm pretty sure the type checker won't be happy about about just sticking a `None | Any` into a `None | str`.

You don't have to use this, but just an example of what I'm talking about:

```suggestion
        chat_template = tokenizer_config.get('chat_template')
        if chat_template is None or isinstance(chat_template, str):
            self.chat_template = chat_template
        else:
            print(f'gguf: warning: Bad type for chat_template field in {tokenizer_config_file!r} - ignoring',
                file = sys.stderr)
```

Pretty minor, but the code would probably be clearer if you moved it up to just before the `for` loop.

**Review by KerfuffleV2 — CHANGES_REQUESTED** [review]

File: gguf-py/pyproject.toml
Comment (KerfuffleV2) [reply gguf-py/pyproject.toml:3]: You definitely may be right. In the past though, we've just bumped the patch version for small changes like adding a metadata field. I don't think the version is a big deal, so it's probably okay either way. I'd suggest asking someone that can give a definitive answer though.

File: gguf-py/gguf/vocab.py
Comment (KerfuffleV2) [reply gguf-py/gguf/vocab.py]: Ahh, there's actually one more thing you need to do and that's set `self.chat_template` to `None` in `__init__`. Otherwise it's possible to end up accessing it when it never got set to something.

**Review by KerfuffleV2 — APPROVED** [review]
This looks good to me. I tested the various cases:

1. `chat_template` field not present in `tokenizer_config.json`
2. Present but with an invalid type
3. Not present

I'll let you use your own discretion for the version thing, if you want to do any further checking before merging.

File: gguf-py/pyproject.toml
Comment (cebtenzzre) [reply gguf-py/pyproject.toml:3]: slaren is right, we should bump the minor version when we add new features, even small ones.

**Pull request merged**
